src/TodoList.tsx

```diff
@@ -0,0 +1,174 @@
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+import { TodoItem } from './utils/TodoItem';
+
+interface TodoListProps {
+  todos: Todo[];
+  setRemoveTodoIsClicked: React.Dispatch<React.SetStateAction<boolean>>;
+  setEditTodoIsClicked: React.Dispatch<React.SetStateAction<boolean>>;
+  tempTodo: Todo | null;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setToggleTodoIsFailed: React.Dispatch<React.SetStateAction<boolean>>;
+  updatingTodosToggle: number[];
+  deletingCompletedTodos: number[];
+  setIsInputDisabled: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const TodoList: React.FC<TodoListProps> = ({
+  todos,
+  setRemoveTodoIsClicked,
+  setEditTodoIsClicked,
+  tempTodo,
+  setTodos,
+  setToggleTodoIsFailed,
+  updatingTodosToggle,
+  deletingCompletedTodos,
+  setIsInputDisabled,
+}) => {
+  const [itemToDelete, setItemToDelete] = useState<number | null>(null);
+  const [itemToToggle, setItemToToggle] = useState<number | null>(null);
+  const [itemToUpdate, setItemToUpdate] = useState<number | null>(null);
+  const [itemToLoad, setItemToLoad] = useState<number | null>(null);
+
+  const toggleTodo = async (todoId: number) => {
+    setItemToToggle(todoId);
+
+    const todoToUpdate = todos.find((todo) => todo.id === todoId);
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const { id, userId, title } = todoToUpdate!;
```

Comment:
The non-null assertion operator (!) should not be used as it can lead to runtime errors. You should check the object is not null before destructuring it.